librtt/Display/Rtt_SpriteObject.cpp

```diff
@@ -100,31 +126,59 @@ SpriteObjectSequence::Create( Rtt_Allocator *allocator, lua_State *L, int index
 	Direction loopDirection = DirectionForString( lua_tostring( L, -1 ) );
 	lua_pop( L, 1 );
 
+	Real time = 0;
+	Real *timeArray = NULL;
+	int numFramesInTimeArray = 0;
+	lua_getfield( L, index, "time" );
+	if ( lua_isnumber(L, -1) )
+	{
+		time = luaL_toreal( L, -1 );
+	}
+	else if ( lua_istable( L, -1 ) )
+	{
+		numFramesInTimeArray += (int) lua_objlen( L, -1 );
+		timeArray = (Real *)Rtt_MALLOC( allocator, numFramesInTimeArray * sizeof( Real ) );
+		for ( int i = 0; i < numFramesInTimeArray; i++ )
```

Comment:
like in here, do `for ( int i = 0; i < numFramesInTimeArray && i<numFrames; i++ )`
And later something like
```
for ( int i = numFramesInTimeArray;  i<numFrames; i++ )
    timeArray[i] = lastGoodValue
```
And do static check and trace if numFramesInTimeArrat!=numFrames

Reply:
Makes sense, latest commit changed the SpriteObjectSequence::Create implementation to only allocate the timeArray once.